apps/dsky_clock/app.js

```diff
@@ -0,0 +1,373 @@
+      setLight('L6','GPS',true);
+      mode_GPS();
+      break;
+    case 4:
+      setLight('L6','ACCEL',true);
+      mode_accel();
+      break;
+    case 5:
+      setLight('L6','HDG',true);
+      mode_compass();
+      break;
+    default:
+      drawMain();
+  }
+  layout.render();
+}
+
+function mode_HRM() {
+  setLight('COMPACTY','',true,Light_COMPACTY);
+  AltDrawTimer = setTimeout( function() {
+    Bangle.setHRMPower(true, 'dsky_clock');
+    let hrm=getHRM();
+    setDATA('R1',hrm.bpm);
+    setDATA('R2',hrm.bpmConfidence);
+    setDATA('R3',getSteps());
+    mode_HRM();
+  }, 5000);
+  Bangle.setHRMPower(false);
+}
+
+function mode_weather() {
+  let weather=getWeather();
+  weather.temp = Math.round(weather.temp-273.15);
+  setDATA('R1',weather.temp);
+  setDATA('R2',weather.hum);
+  setDATA('R3',weather.code);
+  setWORD('NOUN',weather.hum);
+  setWORD('VERB',weather.rain);
+}
+
+function mode_compass() {
+  AltDrawTimer = setTimeout ( function() {
+    setLight('COMPACTY','',true,Light_COMPACTY); //isCompassOn seems to be incorrect?
+    Bangle.setCompassPower(1);
+    let compass=Bangle.getCompass();
+    setDATA('R1',compass.heading);
+    setDATA('R2');
+    setDATA('R3');
+    mode_compass();
+  }, 200);
+  Bangle.setCompassPower(0);
+}
+
+function mode_GPS() {
+  setLight('COMPACTY','',true,Light_COMPACTY);
+  AltDrawTimer = setTimeout( function() {
+    Bangle.setGPSPower(1,'dsky_clock');
+    let gps=Bangle.getGPSFix();
+    setWORD('NOUN',gps.fix);
+    setWORD('VERB',gps.satellites);
+    setDATA('R1',gps.lat);
+    setDATA('R2',gps.lon);
+    setDATA('R3',gps.speed);
+    mode_GPS();
+  }, 5000);
+  Bangle.setGPSPower(0);
+}
+
+function mode_accel() {
+  AltDrawTimer = setTimeout( function() {
+    setLight('COMPACTY','',isActive(),Light_COMPACTY);
+    let accel=Bangle.getAccel();
+    setDATA('R1',accel.x);
+    setDATA('R2',accel.y);
+    setDATA('R3',accel.z);
+    mode_accel();
+  }, 100);
+}
+
+//////////// Main
+
+var drawTimeout;
+// schedule a draw for the next minute
+function queueDraw() {
+  if (drawTimeout) clearTimeout(drawTimeout);
+  drawTimeout = setTimeout(function() {
+    drawTimeout = undefined;
+    mode = 0;
+    if (AltDrawTimer) clearTimeout(AltDrawTimer);
+    drawMain();
+  }, 60000 - (Date.now() % 60000));
+}
+
+Bangle.on('lock',on=>{
+  mode = 0;
+  drawMain(); // draw immediately
+  });
+Bangle.on("message",function() { setLight('COMPACTY','',isActive(),Light_COMPACTY);});
+Bangle.on('charging',drawMain);
+NRF.on('connect',function() { setLight('L3','BT',!isBTConnected(),Light_warn); });
+NRF.on('disconnect',function() { setLight('L3','BT',!isBTConnected(),Light_warn); });
+
+Bangle.on('swipe', function(directionLR) {
+  if (directionLR == 1) {
+    mode=mode-1;
+  }
+  if (directionLR == -1) {
+    mode=mode+1;
+  }
+  if (mode < 0 ) { mode=5; }
+  mode=(mode % 6);
+  drawAlt(mode);
+});
+
+g.clear();
+draw_bg();
+
+// Show launcher when middle button pressed
+Bangle.setUI({
```

Comment:
I wonder if it would be better to go without Bangle.setUI and just work with the layout library. But I'm not sure myself.

It doesn't make the button open the launcher for me. Maybe that is a conflict with the layout library?

Reply:
Yes, looks like it breaks when I set layout.setUI() from the drawMain function.  Commented it out for now.